docs/Style.thy

```diff
@@ -298,6 +298,64 @@ text \<open>
   approach it with a breadth-first approach. Since the default isabelle strategy is depth-first,
   prefers (or defers) will be needed, e.g. corres proofs.\<close>
 
+subsection \<open>Using by\<close>
+
+text \<open>
+  When all subgoals of a proof can be solved in one apply statement, use `by`.\<close>
+
+lemma
+  "True"
+  by simp
+
+lemma
+  "X"
+  apply (subgoal_tac "True")
+   prefer 2
+   subgoal by blast
+  apply (subgoal_tac "True")
+   prefer 2
+   subgoal
+     by blast \<comment> \<open>for longer tactic invocations\<close>
```

Comment:
It wasn't immediately obvious to me that longer here was referring to the textual length of the tactic (instead of how long it took to execute). How about something like the following?
```suggestion
     by blast \<comment> \<open>for tactic invocations that do not fit within a single line\<close>
```

Reply:
point taken, but it's also for stuff that would overflow the line if we tried to put it all on the same line, you know like `have "fdlksfsd dlkfjsdlkf jskldj" by (simp add: long list here)` can still be wrapped as:
```
have "fdlksfsd dlkfjsdlkf jskldj"
  by (simp add: long list here)
```
thoughts on a phrase that would work better but cover both thoughts?

Reply:
Hmm, it's not completely clear to me what distinction you are making, but would something like for tactic invocations that would overflow the line work?

Reply:
I am trying to distinguish when the whole assembly (have ... + by ...) won't fit on one line due to the tactic being too long textually, as opposed to the tactic itself not fitting on one line.
I think your latest version does meet that requirement, so I'll change it.

Reply:
done. makes more sense now?

Reply:
I think so, but we should probably also update the similar comment in the following example for using by to replace two apply statements.

Reply:
thanks! hopefully done that too now. if you see anything else, let me know